### PR TITLE
refactor: App to functional component

### PR DESCRIPTION
**Pull request by JF-Cozy**
This PR is a part [2] of multiple PR about the new sort logic in the application. Tests will maybe failed and probably don't after merging previous PR and rebasing this one.

- Refactor App to use functional component / hooks
- (*) Extract the main query to new file
- Cleanup browser index (app init) and store configuration

(*) the Query is still not using hooks. It is the very first refactor, the hook will comes in an other PR

### CODE REVIEW

File: src/components/ContentWrapper.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          $exists: false
+        },
+        _id: {
+          $gt: null
+        }
+      })
+      .indexFields(['_id'])
+
+  return (
+    <Query query={query(client)}>
```
Comment (Crash--) [inline src/components/ContentWrapper.jsx:26]: You can simply <Query query={query}> and `client` will be automatically injected. No need to get it through useClient for instance

File: src/components/ContentWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const ContentWrapper = () => (
+  <Query
+    query={client =>
+      client
+        .all(DOCTYPE_CONTACTS)
+        .include(['accounts'])
+        .where({
+          trashed: {
+            $exists: false
+          },
```
Comment (Crash--) [inline src/components/ContentWrapper.jsx:18]: you can change the selector here too

File: src/store/configureStore.js [hunk 3]
```diff
@@ -8,40 +8,35 @@ import {
...
-    middlewares.push(getReporterMiddleware(cozyClient))
-  } */
 
   const store = createStore(
-    reducers,
+    combineReducers({
+      appReducers,
+      cozy: client.reducer()
+    }),
+    persistedState,
```
Comment (Crash--) [inline src/store/configureStore.js:39]: why `persistedState` is not in `combineReducers` anymore? We're changing the behavior by doing that. It should not be tagged as refactor if this is the case

File: src/targets/browser/index.jsx [hunk 4]
```diff
@@ -1,121 +1,59 @@
...
+  const client = getClient()
   const persistedState = {}
   const store = configureStore(
     client,
     polyglot.t.bind(polyglot),
     persistedState
   )
+
   /** I don't know why I need to for this... But if I don't it seems that
    * we have a race between configureStore and initCozyBar resulting in
```
Comment (Crash--) [inline src/targets/browser/index.jsx:34]: initCozyBar doesn't exist anymore

**Comment by Crash--** [comment]
Tests are red. 

Also, if I had to be a pain in the ass I'd say you should have made a commit for the hook translate, one for the breakpoint etc.

**Comment by JF-Cozy** [comment]
> Tests are red.

Yes I know, as I said in the PR description `Tests will maybe failed and probably don't after merging previous PR and rebasing this one.`

> Also, if I had to be a pain in the ass I'd say you should have made a commit for the hook translate, one for the breakpoint etc.

Hmmm hooks like those just add a `const` and remove entries in the `export`. And a wrapper in the `breakpoint` case... so just few changes, but ok next time I will do it by hooks.

File: src/store/configureStore.js (hunk 3 above)
Comment (JF-Cozy) [reply src/store/configureStore.js]: Damned my bad, it's a copy/past from Banks

**Comment by Crash--** [comment]
> Hmmm hooks like those just add a const and remove entries in the export. And a wrapper in the breakpoint case... so just few changes, but ok next time I will do it by hooks.

It's just way easier to review commit by commit with one commit dedicated to one thing.

**Comment by JF-Cozy** [comment]
Can be merged, previous PR https://github.com/cozy/cozy-contacts/pull/661 was merged

**Comment by JF-Cozy** [comment]
> > Tests are red.
> 
> Yes I know, as I said in the PR description `Tests will maybe failed and probably don't after merging previous PR and rebasing this one.`. So now previous PR is merged, this one rebased and it's ok.
> 

My mistake it's not in this PR :( I add it right now, so my apologize

**Review by Crash-- — APPROVED** [review]

**Pull request merged**